lib/blockchain.py

```diff
@@ -315,16 +325,75 @@ def segwit_support(self, N=144):
         h = self.local_height
         return sum([self.BIP9(h-i, 2) for i in range(N)])*10000/N/100.
 
+    def get_suitable_block_height(self, suitableheight):
+
+	#In order to avoid a block in a very skewed timestamp to have too much
+	#influence, we select the median of the 3 top most block as a start point
+	#Reference: github.com/Bitcoin-ABC/bitcoin-abc/master/src/pow.cpp#L201
+	blocks2height=suitableheight
+	blocks1height=suitableheight-1
+	blocksheight=suitableheight-2
+	blocks2 = self.read_header(blocks2height)
+	blocks1 = self.read_header(blocks1height)
+	blocks = self.read_header(blocksheight)
+	if (blocks['timestamp'] > blocks2['timestamp'] ):
+		blocksheight,blocks2height = blocks2height,blocksheight
+	if (blocks['timestamp'] > blocks1['timestamp'] ):
+		blocksheight,blocks1height = blocks1height,blocksheight
```

Comment:
This doesn't look right to me.  Note that in the original code the action of the 1st if can affect the evaluation of the 2nd if.  This isn't the case in your code.

Why not make blocks an array and cut-n-paste my code from mining.py we used for the simuls?

Reply:
https://github.com/kyuupichan/difficulty/blob/master/mining.py <- this mining.py?

Reply:
@kyuupichan which algo in mining.py is it? cw-144?

Reply:
cw-144.  freetrader is still reviewing if it is correctly implemented.  I just merged a PR of his as I'd only done the 3-block sorting in deadalnix's orginal algo, and not the most recent one

Reply:
@kyuupichan i see your point .  should i wait anyway for FT ?

Reply:
From what I've read the sorting network is identically implemented in mining.py as in the C++ code.

Reply:
I can confirm it's implemented identically in ABC, mining.py and fyookball/lib/blockchain.py

Reply:
Hm, looks like I'll have to make some testdata that doesn't have monotonically increasing timestamps...

Reply:
@kyuupichan is correct.  This needs to be done against an array.

Reply:
this code works.  the key is 'block_height' @schancel